test/e2e/main_test.go

```diff
@@ -78,19 +80,46 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+type InstallTestCase struct {
+	Name      string
+	ExpectErr bool
+	TestFunc  func(t *testing.T, testReportingFramework *reportingframework.ReportingFramework)
+}
+
 func TestInstallMeteringAndReportingProducesData(t *testing.T) {
 	testInstallConfigs := []struct {
 		TargetPods                int
 		Name                      string
 		MeteringOperatorImageRepo string
 		MeteringOperatorImageTag  string
 		MeteringConfigSpec        metering.MeteringConfigSpec
+		InstallSubTests           []InstallTestCase
 	}{
 		{
-			Name:                      "HDFSInstall",
+			Name:                      "DefaultMeteringConfigProducesError",
 			TargetPods:                defaultTargetPods,
 			MeteringOperatorImageRepo: meteringOperatorImageRepo,
 			MeteringOperatorImageTag:  meteringOperatorImageTag,
+			InstallSubTests: []InstallTestCase{
+				{
+					Name:      "testReportingProducesData",
+					TestFunc:  testReportingProducesData,
```

Comment:
I think this TestCase doesn't really make a lot of sense. Should `ExpectErr` maybe be outside of InstallSubTests`? Also I think we need a way to verify what the error is. Like we should maybe change `ExpectErr` to a string which we check exists inside the error returned by install.

Reply:
It's also semi-weird that ExpectErr` is in `InstallTestCase` because the install failing is mostly about the MeteringConfig and will never differ between different InstallTestCases. This makes me believe that maybe we need to make `InstallSubTests` into a non-slice and move ExpectErr out.

Reply:
Depending on the implementation of `InstallSubTests`, we can omit this section entirely and just pass an empty `MeteringConfig.Spec` object, e.g.
```go
rf, err := deployerCtx.Setup()
if testCaseExpectErr {
	require.NotNil(t, err, "(%s) expected an error when deploying metering", testFunction.Name)
	require.Equal(t, err, testCaseExpectErrMsg, "(%s) expected the error returned from ")
} else {
	require.NoError(t, err, "(%s) deploying metering should produce no error", testFunction.Name)
}

if rf != nil && testInstallFunction != nil {
	(run the reporting tests)
	...
}
```

If we still want to support testing multiple functions per a `MeteringConfig` configuration, this would be more difficult to remove as we iterate over that list of functions, install metering, then run reporting tests.